src/devices/bus/a2bus/grapplerplus.cpp

```diff
@@ -0,0 +1,234 @@
+	device_t(mconfig, type, tag, owner, clock),
+	device_a2bus_card_interface(mconfig, *this),
+	m_dsw1(*this, "DSW1"),
+	m_ctx(*this, "centronics"),
+	m_ctx_data_in(*this, "ctx_data_in"),
+	m_ctx_data_out(*this, "ctx_data_out"),
+	m_rom(*this, "grapplerplus_rom"),
+	m_started(false), m_ack(0), m_busy(0), m_perror(0), m_select(0), m_irqbit(0), m_irqenable(false), m_timer(nullptr)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void a2bus_grapplerplus_device::device_start()
+{
+	m_timer = timer_alloc(0, nullptr);
+	m_timer->adjust(attotime::never);
+
+	save_item(NAME(m_started));
+	save_item(NAME(m_ack));
+	save_item(NAME(m_busy));
+	save_item(NAME(m_perror));
+	save_item(NAME(m_select));
+	save_item(NAME(m_irqbit));
+	save_item(NAME(m_irqenable));
+	save_item(NAME(m_rombank));
+}
+
+void a2bus_grapplerplus_device::device_reset()
+{
+	m_started = true;
+	m_ack = 0;
+	m_irqenable = false;
+	m_irqbit = false;
+	lower_slot_irq();
+	m_timer->adjust(attotime::never);
+
+	// set initial state of the strobe line
+	clear_strobe();
+}
+
+void a2bus_grapplerplus_device::device_timer(emu_timer &timer, device_timer_id tid, int param, void *ptr)
+{
+	clear_strobe();
+}
+
+/*-------------------------------------------------
+    read_cnxx - called for reads from this card's cnxx space
+-------------------------------------------------*/
+
+uint8_t a2bus_grapplerplus_device::read_cnxx(uint8_t offset)
+{
+	m_rombank = 0; // reads to cnxx will reset rom bank to bank 1
+	return m_rom[(offset & 0xff)];
+}
+
+/*-------------------------------------------------
+    read_c0nx - called for reads from this card's c0nx space
+-------------------------------------------------*/
+
+uint8_t a2bus_grapplerplus_device::read_c0nx(uint8_t offset)
+{
+	uint8_t printertype = ~(m_dsw1->read()) & 0x0e;
+
+	return  ((m_irqbit ? 1 : 0) << 7) |
+			(BIT(printertype,1) << 6) |
+			(BIT(printertype,2) << 5) |
+			(BIT(printertype,3) << 4) |
+			((m_busy ? 1 : 0)   << 3) |
+			((m_perror ? 1 : 0) << 2) |
+			((m_select ? 1 : 0) << 1) |
+			((m_ack ? 1 : 0)    << 0);
+}
+
+uint8_t a2bus_grapplerplus_device::read_c800(uint16_t offset)
+{
+	return m_rom[offset + (m_rombank==0 ? 0x0 : 0x800)];
+}
+
+/*-------------------------------------------------
+    write_c0nx - called for writes to this card's c0nx space
+-------------------------------------------------*/
+
+void a2bus_grapplerplus_device::write_c0nx(uint8_t offset, uint8_t data)
+{
+	switch (offset)
+	{
+		case 0: // Output data and send a strobe
+			m_ctx_data_out->write(data);
+			start_strobe();
+			break;
+
+		case 1: // Set rom bank to bank 2
+			m_rombank=1;
+			break;
+
+		case 2: // Reset interrupt request and IRQ data bit
+			m_irqenable = false;
+			m_irqbit = false;
+			break;
```

Comment:
If this is supposed to acknowledge an interrupt request, you need to deassert it as well.